### PR TITLE
feat(app): i18n tweaks

### PR DESCRIPTION
**Pull request by eddieantonio**
Improves localzation by:

 - implementing `determinePreferredLocale()`, which finds the first matching locale in common with our translations
 - moving translation files to TypeScript — it's already helped find an erroneous duplicate key!
 - accessing all locales from `./langs` instead of importing each locale manually.

### CODE REVIEW

File: src/app/lang/en.ts [hunk 1]
```diff
@@ -1,6 +1,8 @@
+import type Localization from "./localization";
+
 export default {
   common: {
-    name: "Predictive Text Studio",
+    app_name: "Predictive Text Studio",
```
Comment (eddieantonio) [inline src/app/lang/en.ts:5]: There were two keys here labeled `name`, and I only realized this after changing this to TypeScript!

File: src/app/pages/LandingPage.svelte [hunk 2]
```diff
@@ -336,7 +336,7 @@
           Predictive Text Studio
         </h1>
         <p class="masthead__branding">
-          designed for
+          {$_('page.main.designed_for')}
```
Comment (eddieantonio) [inline src/app/pages/LandingPage.svelte:339]: This placeholder was missing.

File: src/app/pages/Languages.svelte [hunk 3]
```diff
@@ -167,7 +167,7 @@
       </a>
       <header class="languages__container--header">
         <div>
-          <h1>{$_('common.name')}</h1>
+          <h1>{$_('common.app_name')}</h1>
           <p>designed for
```
Comment (ChrisChrisLoLo) [inline src/app/pages/Languages.svelte]: Do we want to replace "designed for" with the following?
```suggestion
          <p>{$_('page.main.designed_for')}
```

**Review by ChrisChrisLoLo — APPROVED** [review]
Good stuff, LGTM! :shipit:

File: src/app/pages/Languages.svelte (hunk 3 above)
Comment (eddieantonio) [reply src/app/pages/Languages.svelte]: Good catch!

**Pull request merged**